File: implement-rand10-using-rand7/solution.py
# The rand7() API is already defined for you.
# def rand7():
# @return a random integer in the range 1 to 7

class Solution:
    def rand10(self):
        arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] # starting values considered
        while len(arr) > 1:
            newarr = []
            for i in arr:
                if rand7() > 3: newarr.append(i) # give each number a "fair chance" to move on
            if not newarr: continue 
            arr = newarr
        return arr[0]

File: implement-rand10-using-rand7/solution.cs
/**
 * The rand7() API is already defined for you.
 * function rand7(): number {}
 * @return a random integer in the range 1 to 7
 */

function rand10(): number {
    let ans: number = Number.MAX_SAFE_INTEGER;
    while (ans >= 40) {
        ans = 7 * (rand7() - 1) + (rand7() - 1);
    }
    return (ans % 10) + 1;
};
